File: engine/model/src/main/scala/tools/packet/message/MultiChat.scala
package tools.packet.message

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class MultiChat(private var _name: String, private var _text: String, private var _mode: Int) extends PacketInput {
  def name: String = _name

  def text: String = _text

  def mode: Int = _mode

  override def opcode(): SendOpcode = SendOpcode.MULTICHAT
}

File: engine/engine-model/src/main/java/tools/packet/message/MultiChat.java
package tools.packet.message;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record MultiChat(String name, String text, Integer mode) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MULTI_CHAT;
   }
}